lib/elixir/lib/uri.ex

```diff
@@ -941,6 +941,28 @@ defmodule URI do
   defp path_to_segments(path) do
     path |> String.split("/") |> Enum.reverse()
   end
+
+  @doc """
+  Appends `query` to the given `uri`.
+
+  The given `query` is not automatically encoded, use `encode/2` or `encode_www_form/1`.
+
+  ## Examples
+
+      iex> URI.append_query(URI.parse("http://example.com/"), "x=1") |> URI.to_string()
+      "http://example.com/?x=1"
+
+      iex> URI.append_query(URI.parse("http://example.com/?x=1"), "x=2") |> URI.to_string()
+      "http://example.com/?x=1&x=2"
+  """
+  @doc since: "1.14.0"
```

Comment:
Nitpick: maybe we can add a bodiless clause so that we don't mess with param names for generated docs:

```suggestion
  @doc since: "1.14.0"
  def append_query(uri, query)
```

Reply:
I replaced bodiless clause with these:

```elixir
  def append_query(%URI{} = uri, query) when is_binary(query) and uri.query in [nil, ""] do
  def append_query(%URI{} = uri, query) when is_binary(query) do
```

WDYT?